docs/source/workflows/data_sources.rst

```diff
@@ -0,0 +1,90 @@
+Data Sources
+=============
+
+Data sources are used by modules to pull data from outside of the AI engine.
+For example, a :doc:`predictor <predictors>` may need to define external training data.
+
+CSV Data Source
+----------------
+
+The :class:`~citrine.informatics.data_sources.CSVDataSource` draws data from a CSV file stored on the data platform and annotates it by mapping each column name to a :class:`~citrine.informatics.descriptors.Descriptor`.
+The file is referenced via a :class:`~citrine.resources.file_link.FileLink`.
+Each FileLink references an explicit version of the CSV file.
+Uploading a new file with the same name will produce a new version of the file with a new FileLink.
+
+The columns in the CSV are extracted and parsed by a mapping of column header names to descriptors.
+Columns in the CSV that are not mapped with a descriptor are ignored.
+
+Assume that a file data.csv exists with the following contents:
+
+.. code::
+
+    Chemical Formula,Gap,Crystallinity
+    Bi2Te3,0.153,Single crystalline
+    Mg2Ge,0.567,Single crystalline
+    GeTe,0.7,Amorphous
+    Sb2Se3,1.15,Polycrystalline
+
+That file could be used as the training data for a predictor as:
+
+.. code:: python
+
+    from citrine.informatics.data_sources import CSVDataSource
+    from citrine.informatics.predictors import SimpleMLPredictor
+    from citrine.informatics.descriptors import RealDescriptor, CategoricalDescriptor, ChemicalFormulaDescriptor
+
+    file_link = dataset.files.upload("./data.csv", "bandgap_data.csv")
+
+    data_source = CsvDataSource(
+        file_link = file_link,
+        column_definitions = {
+            "Chemical Formula": ChemicalFormulaDescriptor("formula"),
```

Comment:
(minor request) it could be useful to add that the key here references the column header, and explicitly state that the descriptor key does not need to match the column header key

Reply:
updated